src/common/roll.py

```diff
@@ -99,7 +136,15 @@ def export(self, config: RollExportConfig):
         midi_file = tempfile.NamedTemporaryFile(suffix=".mid")
         midi_wav_file = tempfile.NamedTemporaryFile(suffix=".wav")
 
-        # Write MIDI file.
+        # Create MIDI data.
+        midi_data = MIDIFile(
+            numTracks=len(instruments),
+            ticks_per_quarternote=self.quantization,
+        )
+        for track, ins in enumerate(instruments):
+            ins.add_notes_to_track(midi_data, track)
+
+        # Write MIDI data to MIDI file.
         with open(midi_file.name, "wb") as fout:
             midi_data.writeFile(fout)  # type: ignore
```

Comment:
```suggestion
        midi_data.writeFile(midi_file)
```
`midi_file` acts as an open file pointer